server/src/conduit.mjs

```diff
@@ -31,45 +31,92 @@ import * as commandLine from './commandLine.mjs';
 
 let heartbeatInterval;    // JS Interval ID for heartbeat feature
 let conduitWs;            // WebSocket used by Conduit to talk to Mock Firebolt
+let conduitWss;
 
 // Will (temporarily) hold Firebolt responses sent via the Conduit socket to a client
 const fireboltResponses = {};  // openRpcMsg.id -> response via Conduit from a real Firebolt on a real device
 
 const conduitSocketPort = commandLine.conduitSocketPort;
 
-const conduitWss = new WebSocketServer({ port: conduitSocketPort });
+if (commandLine.conduit) {
+  conduitWss = new WebSocketServer({ port: conduitSocketPort });
 
-conduitWss.on('error', (error) => {
-  console.log('Conduit WebSocket Server Error:');
-  console.log(error.message);
-});
+  conduitWss.on('error', (error) => {
+    console.log('Conduit WebSocket Server Error:');
+    console.log(error.message);
+  });
+
+  // Send pings from server to client to track whether client is alive
+  // Terminates conduitWs if client isn't alive
+  heartbeatInterval = setInterval(function ping() {
+    conduitWss.clients.forEach(function each(conduitWs) {
+      // Note the form of the expression; 'clientIsAlive' might not be present 1st time thru
+      if ( conduitWs.clientIsAlive === false ) {
+        console.log('Heartbeat processing discovered a dead client; terminating socket');
+        return conduitWs.terminate();
+      }
+
+      conduitWs.clientIsAlive = false;
+      const oConduitMsg = {
+        from: 'mock-firebolt',
+        type: 'PING-FROM-SERVER',
+        data: undefined
+      };
+      const conduitMsg = JSON.stringify(oConduitMsg);
+      conduitWs.send(conduitMsg);
+      //console.log('Conduit WebSocket sent a ping message to client');
```

Comment:
This can be comment rather than commenting console.log line.

Reply:
Addressed